src/bit_array.cr

```diff
@@ -32,6 +32,17 @@ struct BitArray
     @bits = Pointer(UInt32).malloc(malloc_size, value)
   end
 
+  def ==(other : BitArray)
+    return false if size != other.size
+    # NOTE: If BitArray implements resizing, there may be more than 1 binary
+    # representation for equivalent BitArrays after a downsize.
+    return LibC.memcmp(@bits, other.@bits, malloc_size) == 0
+  end
+
+  def ==(other)
```

Comment:
Why this? An equality check against anything else shouldn't gives a compile error?